geps/gep-2907/index.md

```diff
@@ -0,0 +1,183 @@
+API to ensure that all our independent proposals related to TLS result in a
+coherent set of APIs. This will result in some adjustments to provisional and
+experimental TLS-related GEPs, specifically [BackendTLSPolicy](/geps/gep-1897)
+and [Client Certificate Verification](/geps/gep-91).
+
+## Goals
+* Define high-level terminology for how we refer to TLS in Gateway API.
+* Define top level fields where TLS configuration can live.
+
+## Non-Goals
+* Add or change fields directly. (This may inspire changes in other GEPs
+  though).
+* Commit to including specific parts of TLS configuration in Gateway API. (This
+  is merely to provide space for future configuration, not a commitment that we
+  will add it to the API.)
+
+## Out of Initial Scope
+There are a variety of related TLS concepts in Gateway API that are not currently
+in scope for this GEP. In the future, this GEP may be expanded to include:
+
+1. Automatic mTLS (often associated with Service mesh)
+1. TLS Passthrough
+1. TLSRoute
+
+
+## Introduction
+
+### Where TLS could be configured
+We have three different places where we might want to configure TLS:
+
+#### A. Gateways
+Config attached at this layer will apply to everything behind a single address
+such as a virtual IP.
+
+#### B. Gateway Listeners
+Each Gateway contains multiple “Listeners”. Each HTTPS Listener in a Gateway
+must have a unique combination of protocol, port, and SNI (Server Name
+Indication). TLS configuration attached at this layer should be [isolated via
+SNI](/guides/implementers/#listener-isolation).
+
+#### C. BackendTLSPolicy
+This policy allows us to attach unique TLS configuration per Backend. Depending
+on the organization, this policy may be owned by the application owner or the
+cluster operator. Note that this configuration will be used by all Gateways
+(potentially from different implementations) that are connecting to the backend.
+
+### "Frontend" and "Backend"
+A guiding principle in this naming is to use consistent naming for “Downstream”
+(1+2) and “Upstream” (3+4), similar to Envoy. To avoid the confusion with what
+is upstream and downstream, Gateway API will use “Frontend” and “Backend”.
+
+1. Frontend: The entity connecting to a Gateway, typically a client application
+   and/or web browser.
+2. Backend: The entity a Gateway is routing traffic to, typically the endpoints
+   behind a Service.
+
+There are essentially 4 different segments that could be relevant to TLS
+configuration in Gateway API:
+
+```mermaid
+flowchart LR
+    Frontend -.->|1. Client Cert| Gateway
+    Gateway -.->|2. Server Cert| Frontend
+    Gateway -.->|3. Client Cert| Backend
+    Backend -.->|4. Server Cert| Gateway
+```
+
+The above diagram depicts these four segments as edges in a graph.
+
+## Proposed Segments
+Note that this does not represent any form of commitment that any of these
+fields or concepts will exist within Gateway API. If or when they do, we propose
+the following naming structure:
```

Comment:
I'd like to summarize the proposed segments, only because when I look at them together they don't seem as unified as they could be:

> Gateway.Listener.TLS.FrontendValidation
> Gateway.Listener.TLS
> Gateway.Spec.BackendTLS.ClientCertificateRef
> Gateway.Spec.BackendTLS.Validation

To me the problem of dealing with Listener is better solved by doing something like this:

> Gateway.Listener.TLS.FrontendValidation
> Gateway.Listener.TLS
> Gateway.Transmitter.TLS.BackendValidation
> Gateway.Transmitter.TLS
 
or even

> Gateway.Listener.TLS.Validation
> Gateway.Listener.TLS
> Gateway.Transmitter.TLS.Validation
> Gateway.Transmitter.TLS

We don't have to use the term `Transmitter`, just something other than `BackendTLS`, which clashes with `BackendTLSPolicy`.  I feel a move toward this pattern presents an easier learning curve for people with memory challenges (like me).

Reply:
Good questions, I've added two separate but related sections to address this:

* Why the inconsistency between Frontend and Backend TLS config on Gateways?
* Why Not Listener level to match FrontendValidation?

I've also added a note that we should reconsider these answers before developing any kind of backend TLS config at a Gateway level. Hope that all works for you.